### PR TITLE
[OPPRO-274] Add the validation for Velox unsupported or mismatched expressions with specific input type.

### PR DESCRIPTION
**Pull request by lviiii**
…th specific input type.

## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala [hunk 1]
```diff
@@ -17,19 +17,44 @@
...
 import org.apache.spark.sql.execution._
 import org.apache.spark.sql.execution.exchange.BroadcastExchangeExec
 import org.apache.spark.sql.types.DecimalType
 
 object ExpressionConverter extends Logging {
-  def replaceWithExpressionTransformer(
-                                        expr: Expression,
-                                        attributeSeq: Seq[Attribute]): Expression =
+
+  def doValidate(blacklist: Map[String, String], expr: Expression): Boolean = {
```
Comment (rui-mo) [inline gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:32]: Could you add some notes/comments for this function to explain the purpose?

File: gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala [hunk 2]
```diff
@@ -17,19 +17,44 @@
...
+        if (inputTypeName.equals(input.dataType.typeName)) {
+          return false
+        }
+      }
+    }
+    true
+  }
+
+  def doValidate(expr: Expression): Boolean =
+    GlutenConfig.enableVeloxBackend() && !doValidate(VELOX_EXPR_BLACKLIST, expr)
```
Comment (rui-mo) [inline gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:51]: Would sth. like `GlutenConfig.getConf.isVeloxBackend` work?

Comment (lviiii) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala]: Yes, it works and I've updated.

File: gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala (hunk 1 above)
Comment (lviiii) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:36]: Done

**Comment by rui-mo** [comment]
LGTM.

File: gluten-core/src/main/scala/io/glutenproject/expression/ConverterUtils.scala [hunk 3]
```diff
@@ -469,4 +476,17 @@ object ConverterUtils extends Logging {
   // Other
   final val ROW_CONSTRUCTOR = "row_constructor"
 
+  // The blacklist for Velox unsupported or mismatched expressions with specific input type,
```
Comment (FelixYBW) [inline gluten-core/src/main/scala/io/glutenproject/expression/ConverterUtils.scala:479]: Can we move the configuration to Velox backends folder?

**Review by FelixYBW — APPROVED** [review]

File: gluten-core/src/main/scala/io/glutenproject/expression/ConverterUtils.scala (hunk 3 above)
Comment (lviiii) [reply gluten-core/src/main/scala/io/glutenproject/expression/ConverterUtils.scala]: Yes, I've moved the configuration to Velox backends.

File: backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxTransformerApi.scala [hunk 4]
```diff
@@ -35,6 +33,35 @@ import org.apache.spark.sql.execution.datasources.velox.DwrfFileFormat
...
+    }
+    true
+  }
+
+  /**
+   * Do validate the expressions based on the specific backend blacklist,
+   * the existed expression will fall back to Vanilla Spark.
+   */
+  override def doValidate(expr: Expression): Boolean =
+    GlutenConfig.getSessionConf.isVeloxBackend && doValidate(VELOX_EXPR_BLACKLIST, expr)
```
Comment (zzcclp) [inline backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxTransformerApi.scala:63]: maybe it doesn't need to check whether it's velox backend？this api is only for velox .

**Review by zzcclp — APPROVED** [review]
LGTM. Only some questions.

**Pull request merged**